### PR TITLE
[#111][PR]Added End-to-End test for Study Hour Estimator

### PR DESCRIPTION
**Pull request by EyalAzimov**
- The test adds 2 new courses, verifies the total study time, and then deletes the courses
- added "data-test" values in different parts of code for Cypress to recognize elements

### CODE REVIEW

File: src/components/Calendar/Event/EditEvent.js [hunk 1]
```diff
@@ -91,7 +91,7 @@ export default function EditEvent(props) {
 
                     <EventForm eventState={eventData} eventStateSetter={setEventData} courseArray={course}/>
 
-                    <div>{recurrenceSelection()}</div>
+                    <div> data-test="recurrence-end-date" {recurrenceSelection()}</div>
```
Comment (kikkiii21) [inline src/components/Calendar/Event/EditEvent.js:94]: data-test should inside the {}?

File: cypress/e2e/SP_29_Course_Study_Hour_Estimator.cy.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        cy.get('[data-test="recurrence-end-date"]').click();
+        cy.get('.MuiDialogActions-root > :nth-child(2)').click();
+        cy.get('#startTime').click().type("11:45:00");
+        cy.get('#endTime').click().type("13:00:00");
+        cy.get('[data-test="addButton"]').click();
+        cy.get('[data-test="Intro Formal Methods for Software Engineering"]').should('exist'); //inside typography of eventDisplay
+        cy.wait(2000);
+    })
+
+    it('Add Course Soen 385', function() {
```
Comment (kikkiii21) [inline cypress/e2e/SP_29_Course_Study_Hour_Estimator.cy.js:27]: seems Sonar cloud doesnt like the repetition of the code, it labels it as duplicate code

File: src/components/Calendar/Event/EditEvent.js (hunk 1 above)
Comment (kikkiii21) [inline src/components/Calendar/Event/EditEvent.js:94]: data-test should inside the {}?

**Review by kikkiii21 — CHANGES_REQUESTED** [review]

File: cypress/e2e/SP_29_Course_Study_Hour_Estimator.cy.js (hunk 2 above)
Comment (EyalAzimov) [reply cypress/e2e/SP_29_Course_Study_Hour_Estimator.cy.js:27]: I will make some functions!

File: src/components/Calendar/Event/EditEvent.js (hunk 1 above)
Comment (EyalAzimov) [reply src/components/Calendar/Event/EditEvent.js]: I got rid of it! Wasn't needed! Thanks

**Review by mnashat — APPROVED** [review]
Great job on the tests! Thanks!

**Pull request closed without merging**